library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphFeaturesTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.tinkerpop;
+
+import org.apache.tinkerpop.gremlin.structure.Graph.Features;
+
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphEdgeFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphGraphFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphVertexFeatures;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraphFeatures.GafferPopGraphVertexPropertyFeatures;
+
+public class GafferPopGraphFeaturesTest {
+    final GafferPopGraph graph = mock(GafferPopGraph.class);
+    final Features features = new GafferPopGraphFeatures();
+
+    @Test
+    public void shouldHaveCertainGafferPopGraphGraphFeatures() {
+        given(graph.features()).willReturn(features);
+        final Features graphFeatures = graph.features();
+
+        assertThat(graphFeatures.graph()).isInstanceOf(GafferPopGraphGraphFeatures.class);
+        assertThat(graphFeatures.graph().supportsTransactions()).isFalse();
+        assertThat(graphFeatures.graph().supportsThreadedTransactions()).isFalse();
+        assertThat(graphFeatures.graph().supportsComputer()).isFalse();
+    }
+
+    @Test
+    public void shouldHaveCertainGafferPopGraphVertexFeatures() {
+        final Object id = "testId";
+        given(graph.features()).willReturn(features);
+
+        final Features vertexFeatures = graph.features();
+
+        assertThat(vertexFeatures.vertex()).isInstanceOf(GafferPopGraphVertexFeatures.class);
+        assertThat(vertexFeatures.vertex().supportsRemoveVertices()).isFalse();
+        assertThat(vertexFeatures.vertex().supportsRemoveProperty()).isFalse();
+        assertThat(vertexFeatures.vertex().willAllowId(id)).isTrue();
```

Comment:
This is the same for all the `willAllowId()` checks but you are only testing if a string type can be used here the test could be expanded to a wider range of types?

Worth noting this might also be an issue with the implementation as the `GafferPopGraphFeatures()` class always returns true no mater what Object type the id is which I'm not sure is quite right as even `null` would be accepted.